synapse/logging/opentracing.py

```diff
@@ -339,6 +339,7 @@ def ensure_active_span_inner_2(*args, **kwargs):
                     "There was no active span when trying to %s."
                     " Did you forget to start one or did a context slip?",
                     message,
+                    stack_info=True,
```

Comment:
Nice---didn't know about this! (`traceback.print_stack` is my go-to)

Reply:
sentry has a bug where each line of some stack traces gets turned into a separate issue. I'm hoping this makes it get filed correctly.

Reply:
that's a synapse/twisted bug rather than a sentry bug. Basically uncaught exceptions end up getting printed to `sys.stderr`, but synapse intercepts things written to `sys.stderr` and sends them back to the logger, at ERROR level, line by line; there they are of course intercepted by the sentry handler.

The correct solution would be to not have uncaught exceptions printed to `sys.stderr`, and just let the interpreter-level uncaught exception hook handle them (where sentry will correctly report them). Unfortunately that's happening within the bowels of Twisted, so the next best solution is to make sure there aren't any uncaught exceptions coming from our code.

Reply:
interesting, thanks for clearing things up!